airbyte-cdk/python/airbyte_cdk/utils/schema_inferrer.py

```diff
@@ -91,10 +102,94 @@ def _clean(self, node: InferredSchema) -> InferredSchema:
                         self._clean(value)
             if "items" in node:
                 self._clean(node["items"])
+
+            # this check needs to follow the "anyOf" cleaning as it might populate `type`
+            if isinstance(node["type"], list):
+                if "null" not in node["type"]:
+                    node["type"] = ["null"] + node["type"]
+            else:
+                node["type"] = ["null", node["type"]]
+        return node
+
+    def _add_required_properties(self, node: InferredSchema) -> InferredSchema:
```

Comment:
Nit: naming gets a bit confusing between this and `add_fields_as_required`. Can you add docstrings with comments for them just to clarify the difference and where they're invoked?